### ISSUE TITLE
Hidden plugins

### ISSUE THREAD

**Issue by skellock** [infinitered/gluegun#122]
We would like support for hidden plugins & commands.  

This is useful if CLIs want to have their own plugins in the gluegun plugin format, but prevent the end user from having to know of them.

You would still be able to call these from `runtime.run({})`.

I recommend adding an object as a 2nd parameter to:

```js
.load('dir', { hidden: true })
.loadDefault('dir', { hidden: true })
.loadAll('dir', { hidden: true })
```

Problem is, `loadAll` already has an optional 2nd parameter for matching sub-directories, so let's upgrade that to:

```js
.loadAll('dir', { hidden: true, matching: 'ignite-*' })`
```

This would be a breaking change, but now is the time!

**Comment by skellock** [comment]
See https://github.com/infinitered/ignite/issues/548 for more info.

**Comment by skellock** [comment]
Implemented in #130
